packages/deployer/internal/contract-helper.js

```diff
@@ -69,10 +70,33 @@ function contractIsModule(contractSourcePath) {
   return source.startsWith(`${hre.config.deployer.paths.modules}${path.sep}`);
 }
 
+/**
+ * Get contract source path absolute location. Will try to look it up on the local
+ * hardat folders, or on node_modules.
+ * @param {string} contractSourcePath contract path to file, e.g.: contracts/modules/SomeModule.sol
+ * @returns {string}
+ */
+function getContractFilePath(contractSourcePath) {
+  // First, try to check if its a contract import from the local project
+  const localSource = path.resolve(hre.config.paths.root, contractSourcePath);
+  if (localSource.startsWith(`${hre.config.paths.sources}${path.sep}`)) {
+    if (!fs.existsSync(localSource)) {
+      throw new Error(`Contract file for ${localSource} not found.`);
+    }
+
+    return path.resolve(hre.config.paths.root, contractSourcePath);
```

Comment:
this is `localSource` already defined a few lines before

Reply:
Fixed